junit-engine-api/src/main/java/org/junit/gen5/engine/support/descriptor/DirectorySource.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2015-2016 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v1.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+
+package org.junit.gen5.engine.support.descriptor;
+
+import static org.junit.gen5.commons.meta.API.Usage.Experimental;
+
+import java.net.URI;
+import java.nio.file.Path;
+import java.util.Objects;
+
+import org.junit.gen5.commons.meta.API;
+import org.junit.gen5.commons.util.Preconditions;
+
+/**
+ * @since 5.0
+ */
+@API(Experimental)
+public class DirectorySource implements FileSystemSource {
+
+	private static final long serialVersionUID = 1L;
+
+	private final Path directoryPath;
```

Comment:
Mmmmm... you're correct: `Path` is in fact not `Serializable`. Looks like we'll have to switch to `File`.

Good catch!